### PR TITLE
Support `nerdctl build` custom outputs alias of type local

### LINKED ISSUE #1422: `nerdctl build` should accept `--output <DIR`> as an alias of `--output type=local,dest=<DIR>`
### Description

`nerdctl build` should accept `--output <DIR`> as an alias of `--output type=local,dest=<DIR>`, as in `docker build`.

### Steps to reproduce the issue

`nerdctl build --output . .`

### Describe the results you received and expected

Received: `error: invalid value .`
Expected: Should work as `--output type=local,dest=$(pwd)`

### What version of nerdctl are you using?

nerdctl version 0.23.0-26-g67c0f59

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

_No response_

### Host information

_No response_

### PR DESCRIPTION
**Pull request by yuchanns**
Fix #1422 

* Also remove call of `testutil.DockerIncompatible` in `TestBuildLocal`

Signed-off-by: Hanchin Hsieh <me@yuchanns.xyz>

### CODE REVIEW

**Comment by yuchanns** [comment]
PTAL @junnplus @AkihiroSuda

File: cmd/nerdctl/build_test.go [hunk 1]
```diff
@@ -202,6 +204,16 @@ COPY %s /`,
 	data, err := os.ReadFile(testFilePath)
 	assert.NilError(t, err)
 	assert.Equal(t, string(data), testContent)
+
+	aliasOutputDir := t.TempDir()
+	testAliasFilePath := filepath.Join(aliasOutputDir, testFileName)
+	base.Cmd("build", "-o", aliasOutputDir, buildCtx).AssertOK()
+	if _, err := os.Stat(testAliasFilePath); err != nil {
+		t.Fatal(err)
```
Comment (AkihiroSuda) [inline cmd/nerdctl/build_test.go:212]: Any reason not to use `assert.NilError` ?

**Review by AkihiroSuda — APPROVED** [review]
Thanks

File: cmd/nerdctl/build_test.go (hunk 1 above)
Comment (yuchanns) [reply cmd/nerdctl/build_test.go:212]: Well, I was planning to fix it.

**Pull request merged**